awscommons/v2/ecs.go

```diff
@@ -0,0 +1,201 @@
+package awscommons
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/gruntwork-io/go-commons/collections"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/logging"
+	"github.com/gruntwork-io/go-commons/retry"
+)
+
+// GetContainerInstanceArns gets the container instance ARNs of all the EC2 instances in an ECS Cluster.
+// ECS container instance ARNs are different from EC2 instance IDs!
+// An ECS container instance is an EC2 instance that runs the ECS container agent and has been registered into
+// an ECS cluster.
+// Example identifiers:
+// - EC2 instance ID: i-08e8cfc073db135a9
+// - container instance ID: 2db66342-5f69-4782-89a3-f9b707f979ab
+// - container instance ARN: arn:aws:ecs:us-east-1:012345678910:container-instance/2db66342-5f69-4782-89a3-f9b707f979ab
+func GetContainerInstanceArns(opts *Options, clusterName string) ([]string, error) {
+	client, err := NewECSClient(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	logger := logging.GetProjectLogger()
+	logger.Infof("Looking up Container Instance ARNs for ECS cluster %s", clusterName)
+
+	input := &ecs.ListContainerInstancesInput{Cluster: aws.String(clusterName)}
+	arns := []string{}
+	// Handle pagination by repeatedly making the API call while there is a next token set.
+	for {
```

Comment:
NIT: consider adding a maximum to the loop iteration to avoid an infinite loop. Logically, the AWS API should prevent such an infinite loop, but it's always good to practice defensive coding to avoid unknown unknowns causing unexpected behaviors.

Reply:
Ah, that's true. I'll take care of this in the next review cycle, putting a todo here for now.